kitsune/wiki/jinja2/wiki/document.html

```diff
@@ -4,7 +4,7 @@
 {% from "wiki/includes/document_macros.html" import contributor_list, document_title, document_messages, document_content, topic_list %}
 {% from "wiki/includes/document_macros.html" import vote_form with context %}
 {% set classes = 'document' %}
-{% set canonical_url = document.get_absolute_url() %}
```

Comment:
The canonical URL for a page should have the locale in it because it should be a URL that returns a 200, not a 301. So I think you can just concat `document.get_absolute_url()` with the `CANONICAL_URL` setting and go with that.